mempool/src/interface/mempool_interface_impl.rs

```diff
@@ -139,7 +139,7 @@ impl MempoolInterface for Mempool {
     fn collect_txs(
         &self,
         tx_accumulator: Box<dyn TransactionAccumulator + Send>,
-    ) -> Result<Box<dyn TransactionAccumulator>, Error> {
+    ) -> Result<Option<Box<dyn TransactionAccumulator>>, Error> {
```

Comment:
Why `Option` rather than adding an arm to `Error`, or having `Result<Result<`? It seems that having some sort of descriptive "error" type is warranted in this case.

Reply:
The problem with errors is the different handling. Returning an error can mean something different. The ideal solution is to have another enum instead of the Option, but I don't care about that right now. In the future we can improve it.